drivers/media/platform/bcm2835/bcm2835-unicam.c

```diff
@@ -1814,12 +2056,22 @@ static int unicam_open(struct file *file)
 	if (!v4l2_fh_is_singular_file(file))
 		goto unlock;
 
+	/*
+	 * Ensure only the image node assigns a ctrl handler.
+	 * struct v4l2_fh gets allocated in v4l2_fh_open() above.
+	 */
+	if (node->pad_id == IMAGE_PAD) {
+		fh = (struct v4l2_fh *)file->private_data;
```

Comment:
Having to cast a private_data pointer seems like the wrong thing to me.
I had a thread with Hans about where to set ctrl_handlers  - https://www.spinics.net/lists/arm-kernel/msg779301.html

quote: You can set ctrl_handlers at three levels: v4l2_device (common to all
video devices), video_device (specific to a single video device), or
for v4l2_fh (specific to that filehandle only).

In this case you would assign it to the video_device:

	dev->vdev.ctrl_handler = &dev->ctrl_handler;
(end quote)

I think you need the same thing here. register_node should be setting ctrl_handler on ```node->video_dev```, not leaving it to be set on the file handle.